### ISSUE TITLE
Error when invoking

### ISSUE THREAD

**Issue by MikaelElkiaer** [Hoffs/omnisharp-extended-lsp.nvim#2]
Hi, trying out this cool plugin as this is a feature I've been missing from VSCode.

However, using nothing more exotic than your example config from the readme, I am still seeing the following error:
```
Error executing vim.schedule lua callback: ...t/omnisharp-extended-lsp.nvim/lua/omnisharp_extended.lua:231: attempt to index field 'params' (a nil value)
```

And this is no matter where I am invoking - internal or external types.

**Comment by Hoffs** [comment]
Omnisharp itself is running and configured? Also what neovim version are you using? Since it could be related to https://github.com/neovim/neovim/pull/15504 which should not be an issue in 0.6 nightly or 0.5.1

**Comment by MikaelElkiaer** [comment]
Yeah, everything should be in order.

Boring `:LspInfo` output:
```
Language client log: /home/me/.cache/nvim/lsp.log
Detected filetype:   cs

1 client(s) attached to this buffer:

Client: omnisharp (id: 1, pid: 400739, bufnr: [12, 12])
 filetypes:       cs, vb
 autostart:       true
 root directory:  /home/me
 cmd:             /usr/bin/omnisharp --languageserver --hostPID nil -s /home/me
```

Boring `nvim -v` output:
```
NVIM v0.5.1
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/build/neovim/src/neovim-0.5.1/build/config -I/build/neovim/src/neovim-0.5.1/src -I/usr/include -I/build/neovim/src/neovim-0.5.1/build/src/nvim/auto -I/build/neovim/src/neovim-0.5.1/build/include
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

**Comment by Hoffs** [comment]
Your hostPID is nil, I see my example didnt include it, checkout https://github.com/neovim/nvim-lspconfig/blob/master/CONFIG.md#omnisharp and setup omnisharp as explained there and then just add that handler.

```lua
local pid = vim.fn.getpid()

local config = {
  handlers = {
    ["textDocument/definition"] = require('omnisharp_extended').handler,
  },
  ...
  cmd = { run_path, '--languageserver' , '--hostPID', tostring(pid) },
}

require'lspconfig'.omnisharp.setup(config)
```

try with fixed pid and see if it starts working now

**Comment by MikaelElkiaer** [comment]
> Your hostPID is nil, I see my example didnt include it, checkout https://github.com/neovim/nvim-lspconfig/blob/master/CONFIG.md#omnisharp and setup omnisharp as explained there and then just add that handler.
> 
> ```lua
> local pid = vim.fn.getpid()
> 
> local config = {
>   handlers = {
>     ["textDocument/definition"] = require('omnisharp_extended').handler,
>   },
>   ...
>   cmd = { run_path, '--languageserver' , '--hostPID', tostring(pid) },
> }
> 
> require'lspconfig'.omnisharp.setup(config)
> ```
> 
> try with fixed pid and see if it starts working now

Good catch. Unfortunately, it did not help. Still the same error.

**Comment by Hoffs** [comment]
And does omnisharp work at all with/without this handler? Like finding references or type hints? For types that are defined internally and not decompiled this is not needed  for "go to definition" to work.

You can also try setting log level to trace with `vim.lsp.set_log_level("trace")` and checking lsp log as seen in `:LspInfo`.

**Comment by MikaelElkiaer** [comment]
Yeah, Omnisharp works just fine, I use it all the time, also for going to internal definitions.

Will try tracing later.

Thanks!

**Comment by MikaelElkiaer** [comment]
I spent a little time digging, but was unable to find anything useful. To me it seems like everything is being invoked as expected, but for some reason is not passed on.

Here's examples of both internal and external types being looked up:
```
[DEBUG][2021-11-10 09:15:44] .../vim/lsp/rpc.lua:395    "rpc.send"  {  id = 2,  jsonrpc = "2.0",  method = "textDocument/definition",  params = {    position = {      ch
aracter = 50,      line = 26    },    textDocument = {      uri = "file:///home/me/Repositories/GitHub/Beehive/Beehive/Services/ContainerService.cs"    }  }}
[DEBUG][2021-11-10 09:15:44] .../vim/lsp/rpc.lua:496    "rpc.receive"   {  id = 2,  jsonrpc = "2.0",  result = {    range = {      end = {        character = 26,
line = 4      },      start = {        character = 17,        line = 4      }    },    uri = "file:///home/me/Repositories/GitHub/Beehive/Beehive/Config/RunConfig.cs"  }
}
[DEBUG][2021-11-10 09:16:21] .../lua/vim/lsp.lua:907    "LSP[omnisharp]"    "client.request"    1   "textDocument/definition"   {  position = {    character = 54,    lin
e = 28  },  textDocument = {    uri = "file:///home/me/Repositories/GitHub/Beehive/Beehive/Services/ContainerService.cs"  }}   <function 1>    nil
[DEBUG][2021-11-10 09:16:21] .../vim/lsp/rpc.lua:395    "rpc.send"  {  id = 3,  jsonrpc = "2.0",  method = "textDocument/definition",  params = {    position = {      ch
aracter = 54,      line = 28    },    textDocument = {      uri = "file:///home/me/Repositories/GitHub/Beehive/Beehive/Services/ContainerService.cs"    }  }}
[DEBUG][2021-11-10 09:16:21] .../vim/lsp/rpc.lua:496    "rpc.receive"   {  id = 3,  jsonrpc = "2.0",  result = {    range = {      end = {        character = 36,
line = 26      },      start = {        character = 15,        line = 26      }    },    uri = "file:///%24metadata%24/Project/Beehive/Assembly/System/Runtime/Symbol/Sys
tem/ArgumentNullException.cs"  }}
```

**Comment by Hoffs** [comment]
If you are invoking with ` require('omnisharp_extended').telescope_lsp_definitions()` and still getting this the issue still to me looks like due to some signature mismatch. If you have time, you could find where the plugin is installed to locally and modify `omnisharp-extended-lsp.nvim/lua/omnisharp_extended.lua` at L231 to 
```
  log.info(ctx)
  local req_from_meta = M.parse_meta_uri(ctx.params.textDocument.uri)
```

then find location of log file with
```
:lua print(vim.api.nvim_call_function('stdpath', {'data'}))
```
+ omnisharp-extended-lsp.nvim.log
and see what it logged when invoking that telescope command. For me it outputs as expected:
```
[INFO  Wed Nov 10 15:33:19 2021] ...d/omnisharp-extended-lsp.nvim/lua/omnisharp_extended.lua:231: {
  client_id = 1,
  method = "textDocument/definition",
  params = {
    position = {
      character = 14,
      line = 19
    },
    textDocument = {
      uri = "file:///../../../...cs"
    }
  }
}
```

**Comment by MikaelElkiaer** [comment]
Thank you so much for your help. Still new to neovim (and vim) and need to learn more about how it all fits together.

Following your very thorough advice, I get the following logs:
```
[INFO  2021-11-10T19:58:31 CET] ...t/omnisharp-extended-lsp.nvim/lua/omnisharp_extended.lua:232: {
  client_id = 1,
  method = "textDocument/definition"
}
[INFO  2021-11-10T20:02:52 CET] ...t/omnisharp-extended-lsp.nvim/lua/omnisharp_extended.lua:232: {
  client_id = 1,
  method = "textDocument/definition"
}
```

Missing params, as the error stated.

**Comment by MikaelElkiaer** [comment]
However, logging here: https://github.com/Hoffs/omnisharp-extended-lsp.nvim/blob/7ea7c029614f0bb6472d8375792836015bf1834d/lua/omnisharp_extended.lua#L266

Yields the expected params:
```
[INFO  2021-11-10T20:09:39 CET] ...t/omnisharp-extended-lsp.nvim/lua/omnisharp_extended.lua:270: {
  position = {
    character = 15,
    line = 11
  },
  textDocument = {
    uri = "file:///home/me/Repositories/GitHub/Beehive/Beehive/Config/RunConfig.cs"
  }
}
```

**Comment by Hoffs** [comment]
Yeah... I just tried with neovim 0.5.1 and got same result.

It seems that while `{params}` was added like 8 months ago to master it was never merged/backported to 0.5 and only exists on master: https://github.com/neovim/neovim/commit/187579fe197ddd844bc0b0979e7f53c646f4235e (and since I have been using nightly I never noticed)

Should be fixed in https://github.com/Hoffs/omnisharp-extended-lsp.nvim/commit/fcd0342c44be63bc2bdb461634f68834e7105823

After the update global handler override is irrelevant for 0.5.1 version and only using the provided functions work. If you are using telescope function - nothing needs to be changed, update the plugin and it should work 🤞 . If you were relying on the built in lsp commands, as in:

```vimscript
<Cmd>lua vim.lsp.buf.definition()<CR>
```

then that function needs to be changed to:

```vimscript
<Cmd>lua require('omnisharp_extended').lsp_definitions()<CR>
```

as this explicitly captures `params` in the closure. I tested both on nightly and 0.5.1 and it worked for me using latest commit version.
